src/pages/private/Practice/PracticeUpdateForm.tsx

```diff
@@ -0,0 +1,141 @@
+import {Button, Col, Form, Input, message, Row} from "antd";
+import ImageUpload from "../../../components/ImageUpload";
+import {FormItem} from "../../../components/Form/FormItem";
+import {validationRules} from "../../../components/Form/validationRules.ts";
+import useForm from "../../../hooks/useForm.tsx";
+import {TPractice} from "../../../hooks/useGetPractice.tsx";
+import {FC} from "react";
+import useUpdatePractice from "../../../hooks/useUpdatePractice.tsx";
+
+type TPracticeUpdateFormProps = {
+    data:TPractice;
+    id:string
+}
+const PracticeUpdateForm:FC<TPracticeUpdateFormProps> = ({data,id})=>{
+    const {form} = useForm()
+    const {mutate,isLoading} = useUpdatePractice(id)
```

Comment:
you can pass this id in hook instead of pass it from business logic

Reply:
I am unable to pass an undefined value of 'id' to the 'doc()' function when creating a document reference for a mutation hook.